util.js

```diff
@@ -1,8 +1,11 @@
 const fs = require('fs')
 const path = require('path')
+const url = require('url')
 
 const resolveFrom = require('resolve-from')
 
+const hasESM = process.versions.node.split('.')[0] >= 14
```

Comment:
This is not true. It would work on Node 12. I think this detection should be more granular: see how it is done in fastify-autoload.

Reply:
It is true that it works on >= 14. 
But it *also* works on >12.17 but not on 13 ;-)

I have seen how its done in fastify-autoload as you mentioned that already near the start of #267 

I tried to avoid the complexity both on code and in the docs by just starting at 14 as:
- 14 was the first version to support from ESM from day 1 without feature flags
- 14 is active LTS and 12 goes into LTS maintenance within a month
- ESM is still experimental anyway

And from your [comment](https://github.com/fastify/fastify-cli/issues/267#issuecomment-716348435) I got the impression that you agreed. 

Do you still want the additional complexity ?

Reply:
ESM is going out of experimental in v12 in the upcoming release and it's already out of experimental in v14. We need the added complexity unfortunately :(

Reply:
You can also check if there is `global.import()` available.

Reply:
I like the global.import() check but the only way that I know is:
```js
let hasESM = false
import('').catch(error => {
    if (error !== 'Not supported') {
        hasESM = true
    }
})
```
as `typeof import === 'function'` returns a syntax error :-(

The downside of the .catch trick is that it sends a warning to stderr on Node 12.18.3:
`(node:20501) ExperimentalWarning: The ESM module loader is experimental.`
Do you have a better way ?

Reply:
added >=12.17.0 <13.0.0